Makefile does not detect virtualenv with Python 3

```
$ virtualenv venv3 -p python3.6
$ . venv3/bin/activate.fish
$ make prepare
A virtualenv must be active.
make: *** [check_venv] Error 1
```

Fix: https://stackoverflow.com/a/42580137
Related to #1780

--- a/Makefile
+++ b/Makefile
@@ -274,7 +274,8 @@ prepare: check_venv
 
 
 check_venv:
-	@$(python) -c 'import sys; sys.exit( int( not hasattr(sys, "real_prefix") ) )' \
+
+	@$(python) -c 'import sys; sys.exit( int( not (hasattr(sys, "real_prefix") or ( hasattr(sys, "base_prefix") and sys.base_prefix != sys.prefix ) ) ) )' \
 		|| ( printf "$(red)A virtualenv must be active.$(normal)\n" ; false )
 
 

